integration_tests/src/main/python/udf_test.py

```diff
@@ -453,11 +458,14 @@ def func(iterator):
                     reason='mapInArrow with barrier mode is introduced by Pyspark 3.5.0')
 @pytest.mark.parametrize('is_barrier', [True, False], ids=idfn)
 def test_map_in_arrow_with_barrier_mode(is_barrier):
+    # The "tc" here can be either a BarrierTaskContext or TaskContext on DB14.3,
+    # not sure any special change made by DB.
+    tc_types = (BarrierTaskContext, TaskContext) if is_databricks143_or_later() else BarrierTaskContext
     def func(iterator):
         tc = TaskContext.get()
         assert tc is not None
         if is_barrier:
-            assert isinstance(tc, BarrierTaskContext)
+            assert isinstance(tc, tc_types)
```

Comment:
Interesting, probably, we should report this issue to databricks. It doesn't make sense to me the TaskContext is not BarrierTaskContext when it's in barrier mode.

Reply:
Maybe DB has integrated them together into a single TaskContext.

Reply:
I have created a follow-on issue for looking into this further if needed https://github.com/NVIDIA/spark-rapids/issues/12477